skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceCallback.scala

```diff
@@ -263,16 +263,16 @@ object RebalanceCallback extends RebalanceCallbackInstances {
     } yield a
   }
 
-  private final case class Pure[+A](a: A) extends RebalanceCallback[Nothing, A]
+  private[consumer] final case class Pure[+A](a: A) extends RebalanceCallback[Nothing, A]
```

Comment:
please avoid relieving restrictions here, you can still achieve required effect…

Reply:
Then the only option for checking would be using Unit as `A` since there's no method to create arbitrary `WithConsumer` and `Error`. I can use any of the existing methods (`RebalanceCallback.commit` for example), but, again, this makes us stay with `Unit` in law-checking.  I'm totally fine with it, just clarifying this.

Reply:
Made ADT members private again. Though, a small "unfairness" in a test is present as a result of this - commented more detailed description in the test

Reply:
I still do not understand your reasoning, could you please point to the code and explain once more 

> here's no method to create arbitrary WithConsumer and Error

if user is limited to public methods and combinators, why should we use anything else?


> but, again, this makes us stay with Unit

concrete example please

Reply:
I agree with you that these constructors should stay private so I made them private again. The only subject left to discuss is this detail about constructing `Arbitrary` for testing (and the "fairness" of it)

Reply:
> If we are OK with this, then this implementation should be sufficient

I think we should be ok with :)